pkg/util/kubernetes/apiserver/controllers.go

```diff
@@ -56,16 +65,31 @@ type ControllerContext struct {
 
 // StartControllers runs the enabled Kubernetes controllers for the Datadog Cluster Agent. This is
 // only called once, when we have confirmed we could correctly connect to the API server.
-func StartControllers(ctx ControllerContext) error {
+func StartControllers(ctx ControllerContext) errors.Aggregate {
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(controllerCatalog))
 	for name, cntrlFuncs := range controllerCatalog {
 		if !cntrlFuncs.enabled() {
 			log.Infof("%q is disabled", name)
 			continue
 		}
-		err := cntrlFuncs.start(ctx)
-		if err != nil {
-			log.Errorf("Error starting %q: %s", name, err.Error())
-		}
+
+		// controllers should be started in parallel as their start functions are
+		// blocking until the informers are sync'ed or the sync period timed-out.
+		// for error propagation we rely on a buffered channel to gather errors
+		// from the spawned goroutines.
+		wg.Add(1)
+		go func(f startFunc) {
+			defer wg.Done()
```

Comment:
minor nit: consistency, passing the ctrl func but not the waitgroup

```suggestion
		        wg.Add(1)
		        cntrlFuncs.start(ctx, errChan)
		        wg.Done()
```

Reply:
I think it is not correct to have the `wg.Add(1)` since the goroutine, because it is not sure that the goroutine will be started before the execution of `wg.wait()` line 89

Reply:
Indeed, scratch the suggestion 👍